woocommerce/woocommerce-gutenberg-products-block
# Remove authors filter from Product Collection block

## What

Remove `authors` filter from Product Collection block.

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/11002

## Why

From original issue https://github.com/woocommerce/woocommerce-blocks/issues/11002:

> During a recent discussion (p1695020614699169-slack-C02FL3X7KR6) among our team, we revisited the relevance and utility of the "Authors" filter currently implemented in the Product Collection block. Our analysis brought us to a few crucial observations that warrant further consideration:

> - Limited Usage Scenarios: In general use, WooCommerce doesn't prominently feature or utilize 'authors' for products. The 'author' field doesn't prominently appear in product creation and editing workflows, and it is noted that it does not have much utility in the block.
> - Alternative Solutions: For merchants wanting to categorize or differentiate products based on the individual who added them, existing functionalities like tags and categories offer an alternative solution.
> - User Experience (UX): There exists a concern that retaining the "Authors" filter might potentially clutter the user interface, thereby possibly detracting from the overall user experience. Considering the filter doesn't cater to a prevalent use-case, its existence might be more of a hindrance than a benefit.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Go to Editor
2. Add Product Collection block
3. Disable "Inherit query from template"
4. Make sure there's no "Authors" filter available in the Inspector Controls
5. Smoke test the Product Collection in the Editor and Frontend

1. Go to Editor
2. Add Products (Beta) block
3. Disable "Inherit query from template"
4. Add couple of product filters (for example Sale Status and Attributes)
5. Click "Upgrade to Product Collection" in Inspectopr Controls
6. Make sure Product Collection preserved the filters and works correctly after migration
7. Save and check the frontend.


## Screenshots or screencast



| Before | After |
| ------ | ----- |
|     <img width="292" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/20098064/8ae5eeea-5555-47f3-8003-37ae348d8db6">   |   <img width="286" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/20098064/4d121f14-8877-43a4-9cb6-93b248cca80f">    |

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Product Collection: remove Authors filter
